imapautofiler/secrets.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import getpass
+
+import keyring
+
+LOG = logging.getLogger('imapautofiler.client')
+
+
+class FixedPasswordSecret:
+    def __init__(self, password):
+        self.password = password
+
+    def get_password(self):
+        return self.password
+
+
+class KeyringPasswordSecret:
+
+    def __init__(self, hostname, username):
+        self.hostname = hostname
+        self.username = username
+
+    def get_password(self):
+        password = keyring.get_password(self.hostname, self.username)
+        if not password:
+            password = getpass.getpass(
+                'Password for {} (will be stored in the system keyring):'
+                .format(self.username)
+            )
+            keyring.set_password(self.hostname, self.username, password)
+
+        return keyring.get_password(self.hostname, self.username)
```

Comment:
Why not just return password here?

Reply:
Because I'd rather be sure that the one that gets sent out is the one in the keyring. It's a cheap operation, and it seems more "correct" this way. I'm fine either way though.

Reply:
That makes sense, thanks for explaining your thought process.